docs/WritingTests.asciidoc

```diff
@@ -170,7 +170,7 @@ sub post_fail_hook {
 
 ==== `post_run_hook`
 
-It is called after `run` regardless of the result of the test run.
+It is called after `run` only if the `run` subroutine completes successfully, without any failures.
```

Comment:
Did you find out what code change changed behaviour? Is a post_run_hook not intended to be run regardless of the test result?

Reply:
I guess we should not change the existing behavior as people might rely on it - especially because in practice it probably makes more sense if either the post run hook *or* the post fail hook is executed (but not both).

Reply:
> Did you find out what code change changed behaviour? Is a post_run_hook not intended to be run regardless of the test result?

@okurz I ran `git blame` on `basetest.pm`  in os-autoinst but couldn't find any change that should've changed the behavior. I think this is the first time @baierjan noticed and reported this behavior. But I am unsure about it.

Reply:
> I guess we should not change the existing behavior as people might rely on it - especially because in practice it probably makes more sense if either the post run hook _or_ the post fail hook is executed (but not both).

afaik you can even achieve both, if you use `die` or fail to assert something inside `post_run_hook` 😉  but that is probably not a practical use.